src/main/java/com/frostwire/jlibtorrent/TorrentStats.java

```diff
@@ -0,0 +1,144 @@
+package com.frostwire.jlibtorrent;
+import com.frostwire.jlibtorrent.alerts.Alert;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * It will start an alertEvent listener to keep track of the download
+ * and upload Speed (bytes/sec). The granularity selected will depend
+ * on the speed in which the results of the speed of the Alert event
+ * listener is retrieved, which can have a delay in milliseconds.
+ * @author haperlot
+ */
+
+public final class TorrentStats {
+
+    private SessionManager sessionManager;
+    private TorrentHandle torrentHandle;
+    private long samplingIntervalInMs;
+    private long maxHistoryInMs;
+    public static final String UPLOAD = "UPLOAD";
+    public static final String DOWNLOAD = "DOWNLOAD";
+    private Queue<Integer> downloadRate = new LinkedList<>();
+    private Queue<Integer> uploadRate = new LinkedList<>();
+    private long tStart; //for collecting sampling interval time
+
+    public TorrentStats(final SessionManager sessionManager, TorrentHandle torrentHandle, long samplingIntervalInMs, long maxHistoryInMs) {
+        this.sessionManager = sessionManager;
+        this.torrentHandle = torrentHandle;
+        this.samplingIntervalInMs = samplingIntervalInMs;
+        this.maxHistoryInMs = maxHistoryInMs;
+        this.tStart = System.currentTimeMillis();
+        startAlertListener();
+    }
+
+    /**
+     * It will start the alert event listener, as soon as the sampling interval
+     * meets the requested time it will save the values on queues. If the queues
+     * reach max size denoted by the maxHistory time, they will poll() the head
+     * element, in our case the oldest inserted.
+     */
+    private void startAlertListener() {
+
+        sessionManager.addListener(new AlertListener() {
+            @Override
+            public int[] types() {
+                int[] a = new int[1];
```

Comment:
Look at this for a pattern of how to use this: https://github.com/frostwire/frostwire/blob/master/common/src/main/java/com/frostwire/bittorrent/BTDownload.java#L732

What are you doing right now creates a lot of pressure in the GC, since it creates small arrays every time.